spec/models/job_spec.rb

```diff
@@ -272,6 +272,49 @@
         end
       end
     end
+
+    describe ".check_jobs_for_timeout" do
+      before(:each) do
+        @job.update_attributes(:state => "active")
+        relation = double("ActiveRecord::Relations")
+        allow(Job).to receive(:in_my_region).and_return(relation)
+        allow(relation).to receive(:where).and_return(relation)
+        allow(relation).to receive(:each).and_yield(@job)
```

Comment:
Can you create one or more jobs in the database that you check afterwards for timeout?  The mocking and expectations are making it hard to tell what's going on.